kernel/kernel-api/src/main/java/io/delta/kernel/types/DataType.java

```diff
@@ -50,5 +50,21 @@ public boolean equivalent(DataType dataType) {
 
     @Override
     public abstract String toString();
+
+    // TODO: Implement proper parsing.
+    public static DataType fromSQL(String json) {
```

Comment:
JsonHandler has a method to parse the schema as `StructType`. We may need to add another API or update the existing API to return `DataType`.